File: tests/benchmark/corpus/cpp/safe/safe_cross_function_dotdot.cpp
// cpp-safe-016: cross-function bool-returning validator with rejection.
#include <fstream>
#include <string>

bool validate_no_dotdot(const std::string &s) {
    return s.find("..") == std::string::npos
        && (s.empty() || (s[0] != '/' && s[0] != '\\'));
}

void handle(const std::string &user_path) {
    if (!validate_no_dotdot(user_path)) return;
    std::ifstream f(user_path);
}

File: tests/benchmark/corpus/ruby/safe/safe_cross_function_dotdot.rb
# ruby-safe-016: cross-function bool-returning validator with rejection.
def validate_no_dotdot(s)
  !s.include?('..') && !s.start_with?('/') && !s.start_with?('\\')
end

def handle(user_path)
  return unless validate_no_dotdot(user_path)
  File.read(user_path)
end
